opennms-webapp-rest/src/main/java/org/opennms/web/rest/v1/TopologiesRestService.java

```diff
@@ -0,0 +1,155 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.web.rest.v1;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.graphdrawing.graphml.GraphmlType;
+import org.opennms.features.graphml.model.GraphML;
+import org.opennms.features.graphml.model.GraphMLEdge;
+import org.opennms.features.graphml.model.GraphMLGraph;
+import org.opennms.features.graphml.model.GraphMLNode;
+import org.opennms.features.graphml.model.GraphMLWriter;
+import org.opennms.features.graphml.model.InvalidGraphException;
+import org.opennms.netmgt.topologies.service.api.OnmsTopology;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyDao;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyEdge;
+import org.opennms.netmgt.topologies.service.api.OnmsTopologyException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+@Path("topologies")
+public class TopologiesRestService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TopologiesRestService.class);
+
+    @Autowired
+    private OnmsTopologyDao m_topologyDao;
+
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getProtocolSupportedString() {
+        return m_topologyDao.getSupportedProtocols().toString();
+    }
+    
+    @GET
+    @Path("count")
+    @Produces(MediaType.TEXT_PLAIN)
+    public String getCount() {
+        return Integer.toString(m_topologyDao.getSupportedProtocols().size());
+    }
+    
+    @GET
+    @Path("{supported-protocol}")
+    @Produces({MediaType.APPLICATION_XML})
+    public Response getGraph(@PathParam("supported-protocol") String supportedProtocol) {
+        if (!m_topologyDao.getSupportedProtocols().contains(supportedProtocol)) {
+            return Response
+                    .status(Response.Status.NOT_FOUND)
+                    .build();                
+        }
+        try {
+            return Response.ok(getGraphMl(supportedProtocol)).build();
+        } catch (InvalidGraphException | OnmsTopologyException e) {
+            LOG.error("Failed to get graphML",e);
+        }
+        
+        return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+
+    // FIXME enrich the tooltip with address speed and attributes
```

Comment:
FIXME

Reply:
let me check I've just merged develop and it seems that there are other conflict with develop branch.....

Reply:
Still here.